File: solutions/scala/bob/2/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String = {
    val str = statement.trim
    if (str.isEmpty) return "Fine. Be that way!"
    
    val isQuestion = str.endsWith("?")
    val isShout = str.exists(_.isLetter) && str == str.toUpperCase

    (isQuestion, isShout) match {
      case (true, true) => "Calm down, I know what I'm doing!" 
      case (_, true) => "Whoa, chill out!"
      case (true, _) => "Sure."
      case _ => "Whatever."
    }
  }
}

File: solutions/python/bob/2/bob.py
def response(hey_bob):
    str = hey_bob.strip()

    if not str:
        return "Fine. Be that way!"

    isquestion = str.endswith("?")
    isshout = any(c.isalpha() for c in str) and str == str.upper()

    match (isquestion, isshout):
        case (True, True):
            return "Calm down, I know what I'm doing!" 
        case (_, True):
            return "Whoa, chill out!"
        case (True, _):
            return "Sure."
        case _:
            return "Whatever."
